### DISCUSSION TITLE
Modeling a resource that has no Create?

### DISCUSSION THREAD

**Discussion by msummers-nr** [aws/aws-cli#7196] [Q&A] [answered]
How do I model a _public_ facing resource that has no Create and always requires a primary identifier? My reading of this line in the [Resource type schema](https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-schema.html#schema-properties-handlers)

`If the resource type doesn't contain create, read, and delete handlers, CloudFormation can't actually provision the resource.`

Is that a resource without a Create is effectively a no-op, is that correct?

I can fake this by modeling both a "primary identifier" field and the required guid field and then return the guid as the primary identifier- is there a "better" way?

The use case here is a tagging api that requires the guid of the entity to be tagged, the tags themselves have no guid.

TIA

**Comment by aBurmeseDev** [comment, marked as answer]
Hi @msummers-nr. This appears to be an issue with CloudFormation rather than the AWS CLI.

I suggest posting in [AWS CloudFormation GitHub repo](https://github.com/aws-cloudformation) for help or consider asking in [re:Post](https://repost.aws/) as well. 

If you have any other AWS CLI related questions, we're here to help. Thanks!
